conda_libmamba_solver/shards.py

```diff
@@ -68,13 +68,14 @@ def _shards_connections() -> int:
     """
     if context.repodata_threads is not None:
         return context.repodata_threads
-    session = CondaSession()
+    # CondaSession() is expensive to create, we just want to find the default
+    # requests poolmanager size.
+    session = CondaSession.__base__() if CondaSession.__base__ else CondaSession()
```

Comment:
Now that I'm looking at this again, I'm actually really confused why this level of object introspection is needed to get the number of concurrent requests allowable. We don't override this in conda, so I'm not sure why it's worth to go through so much trouble here.

Alternatively, we could just avoid `CondaSession` altogether and just create a normal `sessions.Session` object to get this information.

Reply:
The idea was that conda might override the number in the future. On the other hand we could just be happy with SHARDS_CONNECTIONS_DEFAULT=10 or context.repodata_threads.